samples/chaincode/cc-tools-demo/README.md

```diff
@@ -0,0 +1,274 @@
+		CCID:     ccid,
+		Address:  address,
+		CC:       cc,
+		TLSProps: *tlsProps,
+	}
+
+	//*CC-Tools Specific code (DO NOT OVERWRITE)*//
+
+}
+
+```
+
+## Set the needed env vars in the docker-compose file
+
+From the code above, we need to set two env variables for the chaincode application to work and use FPC and chaincode-as-a-service (CCAAS). One way to do this is to go to `$FPC_PATH/samples/deployment/test-network/docker-compose.yml` and edit both `ecc.peer0.org1.example.com` and `ecc.peer0.org2.example.com` environment block to have 
+```yaml
+- RUN_CCAAS=true
+- FPC_ENABLED=true
+```
+
+
+## Building FPC Go Chaincode
+
+Create a `Makefile` (i.e., `touch $FPC_PATH/samples/chaincode/cc-tools-demo/Makefile`) with the following content:
+
+```Makefile
+TOP = ../../..
+include $(TOP)/ecc_go/build.mk
+
+CC_NAME ?= fpc-cc-tools-demo
+```
+
+Please make sure that in the file above the variable `TOP` points to the FPC root directory (i.e., `$FPC_PATH`) as it uses the `$FPC_PATH/ecc_go/build.mk` file.
+
+**Note**: In our case, we need to change the build command in the `$FPC_PATH/ecc_go/build.mk` file at the `ecc` target to be `ego-go build $(GOTAGS) -o $(ECC_BINARY)` instead of `ego-go build $(GOTAGS) -o $(ECC_BINARY) main.go`
+
+
+In `$FPC_PATH/samples/chaincode/cc-tools-demo` directory, to build the chaincode and package it as docker image, execute:
+
+```bash
+make
+```
+Note: this command runs inside the FPC dev environment and not your local host.
+
+**Note**: If you faced this error:
+```bash
+/project/pkg/mod/github.com/hyperledger-labs/cc-tools@v1.0.1/mock/mockstub.go:146:22: cannot use stub (variable of type *MockStub) as shim.ChaincodeStubInterface value in argument to stub.cc.Init: *MockStub does not implement shim.ChaincodeStubInterface (missing method PurgePrivateData)
+```
+This is because a minor difference between the `ChaincodeStubInterface` used in the cc-tools `Mockstub` as it's missing the `PurgePrivateData` method. 
+To solve this, run `go mod vendor` in the `$FPC_PATH` root directory to download all used packages and go to the file of the error to add the missing method there. 
+```bash
+nano $FPC_PATH/vendor/github.com/hyperledger-labs/cc-tools/mock/mockstub.go
+```
+add the following function
+```go
+ 	// PurgePrivateData ...
+ 	func (stub *MockStub) PurgePrivateData(collection, key string) error {
+ 		return errors.New("Not Implemented")
+ 	}
+
+```
+
+After building, you can check that the `fpc/fpc-cc-tools-demo` image exists in your local docker registry using:
+
+```bash
+docker images | grep fpc-cc-tools-demo
+```
+
+## Time to test!
+
+Next step is to test the chaincode by invoking transactions, for which you need a basic Fabric network with a channel.
+We will use the test network provided in [`$FPC_PATH/samples/deployment/test-network`](../../deployment/test-network).
+To invoke the chaincode, we will use the `simple-cli` application in [`$FPC_PATH/samples/application/simple-cli-go`](../../pplication/simple-cli-go).
```

Comment:
don't we use your cool shell script to test?

Reply:
Also I don't see where you explain how to use `samples/application/cc-demo` here?!

Reply:
Done